core/src/main/scala/cr/pulsar/Pulsar.scala

```diff
@@ -30,8 +32,17 @@ object Pulsar {
     * It will be closed once the client is no longer in use or in case of
     * shutdown of the application that makes use of it.
     */
-  def create[F[_]: Sync](url: PulsarURL): Resource[F, T] =
+  def create[F[_]: Sync](
+      url: PulsarURL,
+      timeout: FiniteDuration
```

Comment:
I think it would be better to introduce something like `sealed abstract class Options`, otherwise we would need to break binary compatibility on every new option we add.

I'd also favor a newtype over a plain `FiniteDuration` given these are two different timeouts.

Reply:
Agreed